data/bundle.go

```diff
@@ -102,10 +104,33 @@ type BundleCommand struct {
 	Executable      []string  `yaml:",omitempty,flow" json:"executable,omitempty"`
 	LongDescription string    `yaml:"long_description,omitempty" json:"long_description,omitempty"`
 	Name            string    `yaml:"-" json:"-"`
+	Triggers        []Trigger `yaml:"triggers,omitempty" json:"trigger,omitempty"`
 	Rules           []string  `yaml:",omitempty" json:"rules,omitempty"`
 	Templates       Templates `yaml:",omitempty" json:"templates,omitempty"`
 }
 
+// Trigger represents the configuration for a command trigger as defined
+// in the bundles/commands/triggers section of the config.
+type Trigger struct {
+	Match string `yaml:"match" json:"match"`
+}
+
+func (c *BundleCommand) MatchTrigger(ctx context.Context, message string) (bool, error) {
+	for _, trigger := range c.Triggers {
+		if c == nil || len(trigger.Match) == 0 {
+			return false, nil
+		}
+		re, err := regexp.Compile(trigger.Match)
```

Comment:
Nit: Could you just add a TODO to pre-cache these patterns?

Reply:
Done.